app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleAdapter.kt

```diff
@@ -110,7 +111,8 @@ class LocaleViewHolder(
         // capitalisation is done using the rules of the appropriate locale (endonym and exonym)
         title.text = locale.getDisplayName(locale).capitalize(locale)
         subtitle.text = locale.displayName.capitalize(Locale.getDefault())
-        icon.isVisible = locale === selectedLocale
+        icon.isVisible =
+            locale == selectedLocale && !LocaleManager.isDefaultLocaleSelected(view.context)
```

Comment:
This should probably be added as a helper to `BaseLocaleViewHolder`

Reply:
Fair, moved it to a method inside base